opt4j-core/src/test/java/org/opt4j/core/common/archive/BoundedArchiveTest.java

```diff
@@ -0,0 +1,203 @@
+	public void addCheckedIndividualDespiteArchiveFull() {
+		Injector injector = Guice.createInjector(new MockProblemModule());
+		IndividualFactory factory = injector.getInstance(IndividualFactory.class);
+
+		Objective o0 = new Objective("o0");
+		Objective o1 = new Objective("o1");
+
+		Individual individual1 = factory.create();
+		Objectives objectivesA0 = new Objectives();
+		objectivesA0.add(o0, 1);
+		objectivesA0.add(o1, 0);
+		individual1.setObjectives(objectivesA0);
+
+		Individual individual2 = factory.create();
+		Objectives objectivesA1 = new Objectives();
+		objectivesA1.add(o0, 0);
+		objectivesA1.add(o1, 1);
+		individual2.setObjectives(objectivesA1);
+
+		BoundedArchive archive = new BoundedArchive(1) {
+			@Override
+			protected boolean updateWithNondominated(Collection<Individual> candidates) {
+				individuals.addAll(candidates);
+				return true;
+			}
+		};
+		archive.addCheckedIndividual(individual1);
+		Assert.assertTrue(archive.contains(individual1));
+		Assert.assertEquals(1, archive.size());
+
+		archive.addCheckedIndividual(individual2);
+	}
+
+	@Test
+	public void addCheckedIndividuals() {
+		Injector injector = Guice.createInjector(new MockProblemModule());
+		IndividualFactory factory = injector.getInstance(IndividualFactory.class);
+
+		Objective o0 = new Objective("o0");
+		Objective o1 = new Objective("o1");
+
+		Individual individual1 = factory.create();
+		Objectives objectivesA0 = new Objectives();
+		objectivesA0.add(o0, 1);
+		objectivesA0.add(o1, 0);
+		individual1.setObjectives(objectivesA0);
+
+		BoundedArchive archive = new BoundedArchive(1) {
+			@Override
+			protected boolean updateWithNondominated(Collection<Individual> candidates) {
+				individuals.addAll(candidates);
+				return true;
+			}
+		};
+		archive.addCheckedIndividuals(Collections.singleton(individual1));
+		Assert.assertTrue(archive.contains(individual1));
+		Assert.assertEquals(1, archive.size());
+	}
+
+	@Test(expected = IndexOutOfBoundsException.class)
+	public void addCheckedIndividualsDespiteArchiveFull() {
+		Injector injector = Guice.createInjector(new MockProblemModule());
+		IndividualFactory factory = injector.getInstance(IndividualFactory.class);
+
+		Objective o0 = new Objective("o0");
+		Objective o1 = new Objective("o1");
+
+		Individual individual1 = factory.create();
+		Objectives objectivesA0 = new Objectives();
+		objectivesA0.add(o0, 1);
+		objectivesA0.add(o1, 0);
+		individual1.setObjectives(objectivesA0);
+
+		Individual individual2 = factory.create();
+		Objectives objectivesA1 = new Objectives();
+		objectivesA1.add(o0, 0);
+		objectivesA1.add(o1, 1);
+		individual2.setObjectives(objectivesA1);
+
+		BoundedArchive archive = new BoundedArchive(1) {
+			@Override
+			protected boolean updateWithNondominated(Collection<Individual> candidates) {
+				individuals.addAll(candidates);
+				return true;
+			}
+		};
+		archive.addCheckedIndividuals(Collections.singleton(individual1));
```

Comment:
Not a change request, just a question: why do you use collections.singleton(individual) instead of just adding the individual?

Reply:
There are to methods to test, one for single objectives and one for a collection of objectives. This one tests the collection.